### ISSUE TITLE
Feature Request: Add right and left bars.

### ISSUE THREAD

**Issue by geommer** [geommer/yabar#35]
Well, if you are new to yabar, I am the author (meh who cares anyway...). I opened this issue to know if anyone would really like to have right or left bars. Currently yabar only supports `top bars` and `bottom bars`. I myself use and will use only topbars. However, I made some reservations in the code if right and left are implemented. I really would like to see some discussion, guides or proposals about this feature if implemented.

I propose:
* The text should automatically be rotated to the center of the screen.
* The bar height in this case is now the width for top and bottom bars and the width is now the height for top and bottom bars so we don't duplicate variables that do the same thing.

**Comment by Airblader** [comment]
I realize that yabar is window manager agnostic, but just as a fair word of warning: i3 does not support vertical docks. :)

**Comment by geommer** [comment]
Thank you @Airblader, I actually thought i3 would automatically resize windows like the top and bottom cases. Since you're experienced in xcb and extensively involved with i3 development, can I ask you what I should expect to be drawn in such case (i.e. vertical bars with i3)?

**Comment by Airblader** [comment]
@geommer i3 will disregard the `_NET_WM_STRUT_PARTIAL` setting and fall back to guessing whether it should be a top or bottom dock. In other words it will resize your dock's window as it sees fit, which, as the window manager, it is allowed to do.

**Comment by geommer** [comment]
Thank you @Airblader, I thought it maybe something related to window_type_dock property.

**Comment by geommer** [comment]
It seems there is no enough interest in such a feature. It will be closed until someone comes up with a better discussion on this issue.

**Comment by flannelhead** [comment]
Hi @geommer, I'm slowly working on this feature as I would like to use it in bspwm where vertical bars are supported. So far I've enabled drawing a bar on the left and right screen edges by selecting the corresponding "position" value in config.

As per your proposal, the roles of width and height are swapped for vertical bars which seems reasonable. However the behaviour of "gap-horizontal" and "gap-vertical" is left intact.

However, I don't quite catch what is meant by "The text should automatically be rotated to the center of the screen." Is the vision here that the text would be vertical as well (rotated by 90 degrees)? I'm thinking of a Windows-like taskbar where the text is horizontal (see e.g. this [image](http://i.stack.imgur.com/ewmC6.jpg)).

**Comment by Vintodrimmer** [comment]
Good day, @geommer.

For what it's worth, I'm extremely interested in having a panel that allows vertical positioning of elements or even multi-line blocks.

My use case is to get rid of the classic always-present top bar in favour of a disappearing right-hand one that basically shows all the information on the keypress.

Right now I'm using conky for something similar, but it's far from an ideal solution. I think dzen2 has this functionality, but I've not tried it yet and heard that it's rather hard to configure.

In other words, if you add such a support to your panel, that would make me one happy user.
